### PR TITLE
feat: Test wrap for ConfirmDialog

### PR DESCRIPTION
**Pull request by caalador**
part of #1371

### CODE REVIEW

File: vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrap.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+     * Get the header of the confirm dialog.
+     *
+     * @return current header of dialog
+     */
+    public String getHeader() {
+        return getComponent().getElement().getProperty("header");
+    }
+
+    /**
+     * Get the header element se to the confirm dialog.
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrap.java:105]:
```suggestion
     * Get the header element set to the confirm dialog.
```

File: vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    @Test
+    void rejectNotAvailable_throwsException() {
+        wrap.open();
+        assertThrows(IllegalStateException.class, wrap::reject);
+    }
+
+    @Test
+    void cancelAvailable_cancelEventClosesDialog() {
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java:51]: do we need a similar test also for reject or is it redundant?

File: vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        wrap.open();
+
+        wrap.cancel();
+
+        Assertions.assertEquals(1, close.get());
+        Assertions.assertFalse(view.dialog.isOpened());
+    }
+
+    @Test
+    void confirmButton_ClosesDialog() {
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java:63]: does it make sense to test also that confirmation listener is invoked when present?

File: vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java (hunk 2 above)
Comment (caalador) [reply vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java:51]: added

File: vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java (hunk 3 above)
Comment (caalador) [reply vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/confirmdialog/ConfirmDialogWrapTest.java:76]: added

**Review by mcollovati — APPROVED** [review]

**Pull request merged**